### PR TITLE
MON-1632 Removing grafana from monitoring stack

### PR DESCRIPTION
**Pull request by prashbnair**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: jsonnet/components/grafana.libsonnet [hunk 1]
```diff
@@ -35,60 +35,18 @@ function(params)
...
-        },
-        port: {
-          targetPort: 'https',
-        },
-        tls: {
-          termination: 'Reencrypt',
-          insecureEdgeTerminationPolicy: 'Redirect',
-        },
-      },
+    config:: {
```
Comment (simonpasquier) [inline jsonnet/components/grafana.libsonnet:38]: It looks like we need to strip out most of the kube-prometheus jsonnet. Could we use github.com/brancz/kubernetes-grafana directly?

**Comment by PhilipGough** [comment]
>  I have removed everything in this PR except the dashboard definitions which we need to retain. Are you suggesting another way of doing the same.

I could be wrong,  but I believe the comment refers to the need to delete (cleanup) existing objects on the upgrade path as well as just "not creating" them on a fresh install which is what this PR does IIUC.

**Comment by prashbnair** [comment]
/retitle MON-1632 Removing grafana from monitoring stack

**Comment by prashbnair** [comment]
@PhilipGough PTAL.

File: pkg/tasks/grafana.go [hunk 2]
```diff
@@ -42,99 +39,14 @@ func NewGrafanaTask(client *client.Client, factory *manifests.Factory, config *m
...
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      grafanaLegacyDataSourceSecretName,
-			Namespace: sds.GetNamespace(),
-		},
-	})
-	if err != nil {
-		return errors.Wrap(err, "deleting legacy Grafana Datasources secret failed")
-	}
-
 	cmdds, err := t.factory.GrafanaDashboardDefinitions()
```
Comment (arajkumar) [inline pkg/tasks/grafana.go:50]: I don't think we even need this. https://github.com/openshift/cluster-monitoring-operator/blob/master/manifests/0000_90_cluster-monitoring-operator_01-dashboards.yaml would serve the console dashboard needs which is be created here https://github.com/openshift/cluster-monitoring-operator/blob/master/hack/build-jsonnet.sh#L37

**Review by arajkumar — CHANGES_REQUESTED** [review]

**Comment by simonpasquier** [comment]
Can you add `TODO` comments for things that need to be removed in 4.12? This covers jsonnet/components/grafana.libsonnet, `pkg/tasks/grafana.go` and this line in `pkg/operator/operator.go`:

https://github.com/openshift/cluster-monitoring-operator/blob/38a487897d146715c734d11e80a2b60e7c374eb0/pkg/operator/operator.go#L609

File: pkg/tasks/grafana.go (hunk 2 above)
Comment (prashbnair) [reply pkg/tasks/grafana.go]: Done

**Comment by prashbnair** [comment]
> Can you add TODO comments for things that need to be removed in 4.12?

Done

File: pkg/tasks/grafana.go [hunk 3]
```diff
@@ -14,12 +14,10 @@
...
+// TODO this file can be be removed in 4.12
 import (
 	"context"
 
-	v1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-
 	"github.com/openshift/cluster-monitoring-operator/pkg/client"
 	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
 	"github.com/pkg/errors"
```
Comment (arajkumar) [inline pkg/tasks/grafana.go:23]: Unused package local var grafanaLegacyDataSourceSecretName should be removed too.

Comment (prashbnair) [reply pkg/tasks/grafana.go:23]: Done

**Review by arajkumar — APPROVED** [review]

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Review by PhilipGough — APPROVED** [review]

**Comment by jan--f** [comment]
/label docs-approved
Adding doc approval as per agreemtent.
Ping @juzhao for qe approval.
Ping @sferich888 for px approval.

**Comment by juzhao** [comment]
@lihongyan1 please help to review and add approve

**Pull request merged**